dip-0001.md

```diff
@@ -0,0 +1,190 @@
+The state of the Genesis block is DEFINED as in BIP009
+
+```C++
+State GetStateForBlock(block) {
+  if (block.height == 0) { 
+    return DEFINED;
+  }
+```
+
+All blocks in the same round have the same state.
+
+```C++
+if ((block.height % 4032) != 0) {
+            return GetStateForBlock(block.parent);
+        }
+```
+
+Otherwise, the next state depends on the previous state:
+
+```C++
+switch (GetStateForBlock(GetAncestorAtHeight(block, block.height - 4032))) {
+```
+
+We remain in the DEFINED state until either we pass the starttime or the timeout. GetMedianTimePast in the code below refers to the median nTime of a block and its 10 predecessors. 
+
+```C++
+  case DEFINED:
+            if (GetMedianTimePast(block.parent) >= timeout) {
+                return FAILED;
+            }
+            if (GetMedianTimePast(block.parent) >= starttime) {
+                return STARTED;
+            }
+            return DEFINED;
+```
+
+After a period in the STARTED state, if we're past the timeout, we switch to FAILED. If not, we tally the bits set, and transition to LOCKED_IN if a sufficient number of blocks in the past period have bit1 set to 1.
+
+Note that a block's state never depends on its own nVersion; only on that of its ancestors.
+
+```C++
+    case STARTED: 
+            if (GetMedianTimePast(block.parent) >= timeout) {
+                return FAILED;
+            }
+            if (thresholdreached) {
+                return LOCKED_IN;
+            }
+            return STARTED;
+```
+
+The protocol change is only in force when the first block is in an ACTIVE state.
+
+## Selection of Parameters
+
+For values of starttime and timeout, we suggest:
+* starttime= 1508025600--turn of Oct 15 at midnight 
+* timeout= 1539561600-- 52 weeks later
+
+This is assuming the code is available mid September.  If the code is available at a later release these numbers should be adjusted.
+
+The selection of a round consisting of 4032 blocks is made based on masternodes not being randomly selected, rather they are in a queue.  As such it can’t treat the sample of masternodes as a simple random sample.  At current time 4032 masternodes would consist of a sample of around 87% of masternodes.  This number is small enough that the sample of masternodes are unique. If this number were larger there would be a chance that some masternodes would be sampled twice while others aren’t sampled at all.  This property is maintained as long as the number of blocks in a round is under 90% of all masternodes.  The number 4032 also has the property of being equal to about a week of blocks.  
+
+## Calculation of thresholdreached
+
+### Determination of thresholdreached
+
+Given a block with block.height % 4032 = 0 we can calculate thresholdreached.  Thresholdreached is determined by bit1 on each version of all blocks in the previous round.  Note the current block is not considered to be in the previous round.  We define thresholdreached to be true when and only when, 
```

Comment:
(format code and thresholdreached with back-ticks)

Given a block with `block.height % 4032 = 0` we can calculate `thresholdreached`.  `thresholdreached` is determined by bit1 on each version of all blocks in the previous round.  Note the current block is not considered to be in the previous round.  We define `thresholdreached` to be true when and only when: